projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,687 @@
+    { id: 2, name: 'The Dark Knight' },
+  ]);
+  fixture.detectChanges();
+
+  const movieNames = fixture.debugElement.queryAll(By.css('p')).map((el) => el.nativeElement.textContent);
+  expect(movieNames).toEqual(['1: Harry Potter', '2: The Dark Knight']);
+});
+```
+
+The test mocks only the properties and methods that are used by the component in that particular test. Even if a Signal Store has additional methods, it is not necessary to mock all of them.
+
+### ng-mocks
+
+ng-mocks is a popular library and can be used to mock the Signal Store as well.
+
+```typescript
+it('should show movies (ng-mocks)', () => {
+  const movies = signal(new Array<Movie>());
+  const loading = signal(false);
+  const moviesStore = MockService(MoviesStore, {
+    movies,
+    loading,
+    load: jest.fn<void, [Signal<string>]>(),
+  });
+
+  TestBed.configureTestingModule({
+    imports: [MoviesComponent],
+    providers: [{ provide: MoviesStore, useValue: moviesStore }],
+  });
+
+  // ...rest as in the previous example
+});
+```
+
+### "Partial Mocking" via Spies
+
+We can also use partial mocking to mock only the `load` method. This approach has the advantage of allowing computeds to function correctly without needing to mock them.
+
+```typescript
+it('should show movies (spy)', () => {
+  TestBed.configureTestingModule({
+    imports: [MoviesComponent],
+    providers: [
+      {
+        provide: MoviesService,
+        useValue: {},
+      },
+    ],
+  });
+
+  const moviesStore = TestBed.inject(MoviesStore);
+  const loadSpy = jest.spyOn(moviesStore, 'load');
+  const fixture = TestBed.createComponent(MoviesComponent);
+
+  fixture.autoDetectChanges(true);
+
+  const studio = loadSpy.mock.calls[0][0];
+  if (studio instanceof Observable || typeof studio === 'string') {
+    throw new Error('Expected signal');
+  }
+
+  const input: HTMLInputElement = fixture.debugElement.query(
+    By.css('input')
+  ).nativeElement;
+
+  expect(studio()).toBe('');
+
+  input.value = 'Warner Bros';
+  input.dispatchEvent(new Event('input'));
+  expect(studio()).toBe('Warner Bros');
+
+  patchState(moviesStore, {
+      movies:
+        [
+          {id: 1, name: 'Harry Potter'},
+          {id: 2, name: 'The Dark Knight'},
+        ]
+    }
+  );
+
+  fixture.detectChanges();
+
+  const movies = fixture.debugElement.queryAll(By.css('p')).map((el) => el.nativeElement.textContent);
+  expect(movies).toEqual(['1: Harry Potter', '2: The Dark Knight']);
+})
+```
+
+This version requires that the test can modify the state, even if it is protected.
+
+We are currently evaluating options for test helpers that allow state modification for protected states without introducing new mocking features that are already covered by existing testing frameworks.
+
+Additionally, there is a community project by Gergely Szerovay that provides a comprehensive mocking library for Signal Stores. You can find more details here: https://www.angularaddicts.com/p/how-to-mock-ngrx-signal-stores
```

Comment:
Same here. To keep this guide concise, we can add the "Community Plugins" page and share community work there.